nuxeo-compound-documents/src/test/java/org/nuxeo/compound/documents/TestCompoundDocumentRest.java

```diff
@@ -43,57 +51,84 @@
+        String data = "{\"entity-type\": \"document\",\"type\": \"" + COMPOUND_FOLDER_DOCTYPE
+                + "\", \"name\": \"myCompoundDocumentFolder\"}";
         try (CloseableClientResponse response = getResponse(RequestType.POST, "path/myCompoundDocument", data)) {
             assertEquals(SC_CREATED, response.getStatus());
         }
         var headers = Map.of(MarshallingConstants.EMBED_ENRICHERS + ".document", SubtypesJsonEnricher.NAME);
         try (CloseableClientResponse response = getResponse(RequestType.GET,
                 "path/myCompoundDocument/myCompoundDocumentFolder", headers)) {
-            assertCompoundResponse(response, "Folderish");
+            assertCompoundResponse(response, Set.of("Folderish"));
         }
     }
 
-    protected void assertCompoundResponse(CloseableClientResponse response, String expectedFacet) throws IOException {
+    protected void assertCompoundResponse(CloseableClientResponse response, Set<String> expectedFacets)
+            throws IOException {
         assertEquals(SC_OK, response.getStatus());
         JsonNode node = mapper.readTree(response.getEntityInputStream());
-        assertEquals(1, node.get("facets").size());
-        assertEquals(expectedFacet, node.get("facets").get(0).asText());
+        JsonNode facets = node.get("facets");
+        assertEquals(expectedFacets.size(), facets.size());
+        facets.forEach(f -> assertTrue(expectedFacets.contains(f.asText())));
         var actualSubtypes = node.get("contextParameters").get("subtypes");
         assertEquals(5, actualSubtypes.size());
-        var allowedSubtypes = Set.of("File", "Picture", "Video", "Audio", "CompoundDocumentFolder");
+        var allowedSubtypes = Set.of("File", "Picture", "Video", "Audio", COMPOUND_FOLDER_DOCTYPE);
         actualSubtypes.forEach(t -> assertTrue(allowedSubtypes.contains(t.get("type").asText())));
     }
+
+    @Test
+    public void testCompoundDocumentUpload() throws IOException {
+        String batchId;
+        try (CloseableClientResponse response = getResponse(RequestType.POST, "upload")) {
+            JsonNode node = mapper.readTree(response.getEntityInputStream());
+            batchId = node.get("batchId").asText();
+        }
+        Builder builder = service.path("upload/" + batchId + "/0")
+                                 .accept(MediaType.APPLICATION_JSON)
+                                 .header("X-File-Name", "test");
+        try (FileInputStream in = new FileInputStream(makeTestZipBlob().getFile());
+                CloseableClientResponse response = CloseableClientResponse.of(builder.post(ClientResponse.class, in))) {
+            assertEquals(Status.CREATED.getStatusCode(), response.getStatus());
+        }
+        try (CloseableClientResponse response = getResponse(RequestType.POST,
+                "upload/" + batchId + "/0/execute/" + FileManagerImport.ID,
```

Comment:
The execution is currently on the batch file, do we want it to be on the whole batch? @troger any thoughts?

Reply:
works as well, updating. Maybe we would like to send more than 1 file in the batch ?

Reply:
I think this is another user story, because we don't know what the product want when multiple files are uploaded in a batch:
- create a compound by file
- create a compound with all the files

Reply:
for readability, I'll repeat a bit of slack here:
I talked about several files assuming you meant several compounds (not a list of files that make a flat compound)

Reply:
did @troger avoid this topic ? Or did he ?

Reply:
we upload only one ZIP here, so either with or without `/0/` is OK for me... I would say without `/0/` - less characters :)